Purchases/SwiftSources/Purchasing/Transaction.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Transaction.swift
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCTransaction) public class Transaction: NSObject {
+
+    let revenuecatId: String
+    let productId: String
+    let purchaseDate: Date
+
+    internal init(transactionId: String, productId: String, purchaseDate: Date) {
+        self.revenuecatId = transactionId
+        self.productId = productId
+        self.purchaseDate = purchaseDate
+        super.init()
+    }
+
+    internal init(with data: Dictionary<String, Any>, productId: String, dateFormatter: DateFormatter) {
+        self.revenuecatId = data["id"] as! String
```

Comment:
we should avoid force casts since they're pretty annoying to debug, the stack trace isn't ideal. 
so I'd do
```
guard let revenueCatId = data["id"] as? String,
    let dateString = data["purchase_date"] as? String,
    let purchaseDate = dateFormatter.date(from: dateString) else { 
    fatalError("couldn't initialize Transaction from dictionary. Reason: unexpected format. Dictionary: \(data).")
}

self.revenuecatId = revenuecatId
self.purchaseDate = purchaseDate
```
You could also get more granular and do multiple guard lets, of course.

Reply:
Oh didn't know you could chain the conditions of the guard let